typescript/sdk/src/token/adapters/FuelTokenAdapter.ts

```diff
@@ -0,0 +1,184 @@
+import { Contract, Provider, TransactionRequest, Wallet } from 'fuels';
+
+import { Address, Domain, Numberish } from '@hyperlane-xyz/utils';
+
+import { BaseFuelAdapter } from '../../app/MultiProtocolApp.js';
+import warpRouteAbi from '../../fuel-types/abis/Warp-route-abi.json' assert { type: 'json' };
+import { MultiProtocolProvider } from '../../providers/MultiProtocolProvider.js';
+import { ChainName } from '../../types.js';
+import { TokenMetadata } from '../types.js';
+
+import {
+  IHypTokenAdapter,
+  ITokenAdapter,
+  InterchainGasQuote,
+  TransferParams,
+  TransferRemoteParams,
+} from './ITokenAdapter.js';
+
+// Interacts with Native Fuel token
+export class FuelNativeTokenAdapter
+  extends BaseFuelAdapter
+  implements ITokenAdapter<TransactionRequest>
+{
+  constructor(
+    public readonly chainName: ChainName,
+    public readonly multiProvider: MultiProtocolProvider<any>,
+    public readonly addresses: { token: Address },
+  ) {
+    super(chainName, multiProvider, addresses);
+  }
+
+  async getBalance(address: Address): Promise<bigint> {
+    const provider = await this.getProvider();
+    const wallet = Wallet.fromAddress(address, provider);
+    const balance = await wallet.getBalance(this.addresses.token);
+    return BigInt(balance.toString());
+  }
+
+  async getMetadata(): Promise<TokenMetadata> {
+    throw new Error('Metadata retrieval not implemented for Fuel tokens');
+  }
+
+  async getMinimumTransferAmount(_recipient: Address): Promise<bigint> {
+    return 0n;
+  }
+
+  async isApproveRequired(
+    _owner: Address,
+    _spender: Address,
+    _weiAmountOrId: Numberish,
+  ): Promise<boolean> {
+    return false;
+  }
+
+  async populateApproveTx(
+    _params: TransferParams,
+  ): Promise<TransactionRequest> {
+    throw new Error('Approve not required for Fuel tokens');
+  }
+
+  async populateTransferTx({
+    weiAmountOrId,
+    recipient,
+  }: TransferParams): Promise<TransactionRequest> {
+    const provider = await this.getProvider();
+    const wallet = Wallet.fromAddress(this.addresses.token, provider);
+    const tx = wallet.createTransfer(recipient, weiAmountOrId.toString());
+    return tx;
+  }
+
+  async getTotalSupply(): Promise<bigint | undefined> {
+    return undefined;
+  }
+}
+
+export class FuelHypNativeAdapter
+  extends BaseFuelAdapter
+  implements IHypTokenAdapter<TransactionRequest>
+{
+  private contract!: Contract;
+  private provider!: Provider;
+
+  constructor(
+    public readonly chainName: ChainName,
+    public readonly multiProvider: MultiProtocolProvider<any>,
```

Comment:
Same with the `any` here I believe
```suggestion
    public readonly multiProvider: MultiProtocolProvider,
```